docker/Dockerfile

```diff
@@ -73,51 +155,98 @@ RUN strip /usr/local/bin/grepperf \
    /usr/local/bin/getnetworkhistory \
    /usr/local/bin/isolatedServer \
    /usr/local/bin/getrewardhistory \
+   /usr/local/bin/buildTxBlockHashesToNums \
 #    /usr/local/bin/zilliqa \
    /usr/local/lib/libSchnorr.so \
    /usr/local/lib/libCryptoUtils.so \
    /usr/local/lib/libNAT.so \
    /usr/local/lib/libCommon.so \
    /usr/local/lib/libTrie.so
 
-FROM zilliqa/zilliqa-base:${ZILLIQA_BASE_VERSION}
+FROM zilliqa/scilla:${SCILLA_VERSION} AS scilla
+
+# run a copy -L to unfold the symlinkes, and strip all exes
+RUN mkdir -p /scilla/0/bin2/ && cp -L /scilla/0/bin/* /scilla/0/bin2/ && strip /scilla/0/bin2/*
+
+FROM ubuntu:20.04
+
+RUN apt-get update \
```

Comment:
Is there an argument that it is better to leave them separate, because the packages required by the build are very unlikely to be the same as the packages required by the runtime and so, to minimise the size of the final image, we should have separate lists of packages?

Obviously, that requires a bit of work to find out what is required for each, but it shouldn't take too long - I'm happy to do that if you don't think its a bad idea?

Otherwise, I guess we could have a separate stage earlier on which installs these packages, used as the base by both the build and runtime stages.

Reply:
The argument is exactly what I mentioned: you're duplicating the same thing twice which we will need to maintain twice and run twice (unless indeed it's cached). Just the maintenance issue is a problem because it's happened several times before that something was added in one place but forgotten in another.

Reply:
But my point is the actual required packages are probably different, so doesn't it make sense to have separate lists?

Reply:
Compare this one and the one later in the file and you'll see that the two lists are almost identical. There's basically a list of basic apt packages that should be installed anyway + extra packages (for dependencies & environment utility) that are needed at a later stage. You're re-installing everything again at a later build stage. Takes both longer, and if you want to remove or add a common package, you'll have to do it twice. Not sure what the misunderstanding is.